enhancements/etcd/automated-backups.md

```diff
@@ -0,0 +1,284 @@
+
+Since it is an implementation detail with no API impact we can start with utilizing the backup script for simplicity and codify it later.
+
+#### Backup schedule
+
+To enforce the specified schedule we can utilize [CronJobs](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/) to run backup pods.
+
+**TODO:** If load balancing between control-plane nodes is a requirement then it needs to be determined how we can achieve that with the Job or Pod spec.
+- The [`nodeSelector`](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodeselector) field would let us choose the group of control-plane/master nodes but not round-robin between them.
+- The [`nodeName`](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodename) field would let us choose the node
+  - More brittle as node names can change and would require writing custom scheduling logic to load balance across nodes
+
+See the Open Questions section for load balancing concerns.
+
+#### Saving the backup
+
+The backup pod needs to be able to save the backup data onto the host filesystem. To achieve this the backup pod can be mounted with a [hostPath Volume](https://kubernetes.io/docs/concepts/storage/volumes/#hostpath).
+
+**TODO:** Does anything prevent us from doing this e.g authorization or security issues.
+
+#### Alerting
+
+TBD
+
+### Risks and Mitigations
+
+TBD
+
+## Design Details
+
+### Open Questions
+
+### Load balancing and IO impact
+- What would be the IO impact or additional load caused by a recurring execution of the backup pod and taking the snapshot?
+  - If significant, should we avoid selecting the leader to reduce the risk of overloading the leader and triggering a leader election?
+  - We can find the leader via the [endpoint status](https://etcd.io/docs/v3.5/dev-guide/api_reference_v3/#message-statusresponse-apietcdserverpbrpcproto) but the leader can change during the course of the backup or we may not have the option of filtering the leader out if the backup pod is unschedulable on non-leader nodes (e.g low disk space). 
+- Other than the leader IO considerations, when picking where to run and save the backup, do we need to load balance or round-robin across all control-plane nodes?
+  - Leaving it to the scheduler to schedule the backup pod (or having the operator save on the node it's running on) is simpler, but we could end up with all/most backups being saved on a single node. In a disaster recovery scenario we could lose that node and the backups saved.
+  - Could be a future improvement if we start with the simpler choice of leaving it to the scheduler.
+
+### Test Plan
+
+Prior to merging the backup controller implementation, an e2e test would be added to openshift/origin to test the validity of the backups generated from the automated backups feature. The pass rate on this test will feature gate the API until we pass an acceptable threshold to make it available by default.
```

Comment:
promoting to "accessible-by-default" will be allowed once there is an acceptable history of passing restore